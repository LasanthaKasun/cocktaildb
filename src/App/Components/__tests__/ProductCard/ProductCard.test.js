import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { ProductCard } from "../..";

it("should render button component", () => {
  const component = renderer.create(
    <ProductCard
      data={{
        strDrinkThumb: "",
        strInstructions: "",
        strDrink: "",
        strCategory: "",
      }}
      handleFavourite={null}
      add="ADD"
    />
  );
  expect(component).toMatchSnapshot();
});

it("should render button with props", () => {
  render(
    <ProductCard
      data={{
        strDrinkThumb: "",
        strInstructions: "",
        strDrink: "",
        strCategory: "",
      }}
      handleFavourite={null}
      add="ADD"
    />
  );
  const productCard = screen.getByTestId("productCard");
  const productCardButton = screen.getByTestId("productCardButton");
  expect(productCard).toHaveTextContent("ADD");
  expect(productCardButton).toHaveTextContent("ADD");
  expect(productCardButton).not.toHaveTextContent("REMOVE");
});
