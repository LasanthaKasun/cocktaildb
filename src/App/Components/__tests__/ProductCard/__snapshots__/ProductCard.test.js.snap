// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button component 1`] = `
<div
  className="max-w-sm max-  border rounded-lg shadow bg-gray-800 border-gray-700 relative"
  data-testid="productCard"
>
  <div
    className="absolute bg-white rounded-full p-2 right-2 top-2 cursor-pointer hover:bg-yellow-400 duration-500 flex gap-2"
    onClick={null}
    title="Add to favourite"
  >
    <img
      alt="add/remove"
      src="favourite_red.png"
      width="25"
    />
    <span
      className="font-semibold"
      data-testid="productCardButton"
    >
      ADD
    </span>
  </div>
  <img
    alt=""
    className="rounded-t-lg"
    src=""
  />
  <div
    className="p-5"
  >
    <h5
      className="mb-2 text-2xl font-bold tracking-tight text-white"
    />
    <h3
      className="mb-2 text-md font-bold tracking-tight text-gray-200"
    />
    <p
      className="mb-3 font-normal text-gray-400"
    />
  </div>
</div>
`;
